src/ICP.h

```diff
@@ -90,10 +90,10 @@ extern Comm::ConnectionPointer icpOutgoingConn;
 extern Ip::Address theIcpPublicHostID;
 
 /// \ingroup ServerProtocolICPAPI
-HttpRequest* icpGetRequest(char *url, int reqnum, int fd, Ip::Address &from);
+HttpRequestPointer icpGetRequest(char *url, int reqnum, int fd, Ip::Address &from);
 
 /// \ingroup ServerProtocolICPAPI
-bool icpAccessAllowed(Ip::Address &from, HttpRequest * icp_request);
+bool icpAccessAllowed(Ip::Address &, const HttpRequestPointer &);
```

Comment:
Please _keep_ the first parameter name. It is very useful to a human reader of this declaration.

```suggestion
bool icpAccessAllowed(Ip::Address &from, const HttpRequestPointer &);
```